### DISCUSSION TITLE
Change order of chromosomes

### DISCUSSION THREAD

**Discussion by lauradmartens** [open2c/cooler#367] [Q&A] [answered]
Hi, I processed my HiC data using HiC-Pro but unfortunately the chromosomes were in a weird order. I was wondering if there is a way to change the order of the chromosomes of my cooler files? Thanks a lot!

**Comment by golobor** [comment]
+1 to a potential feature request! My colleague had the same problem a week ago.

**Comment by nvictus** [comment, marked as answer]
Because the internal structure of the file is chromosome order sensitive, it does require exporting and reloading data.

For a single cooler, a simple bash one-liner can reorder chromosomes:

```bash
cooler dump --join {input.cool} | cooler load --format bg2 {new.chrom.sizes}:{binsize} - {output.cool}
```
That will re-encode the genomic bins according to the order in the new chrom sizes file. 

To transfer balancing weights or to transfer multiple coolers to make a new mcool requires additional manual steps... I guess this is what the proposed feature request would handle. 

As a simpler alternative for now, you can re-balance and zoomify the converted base resolution cooler with `cooler zoomify --balance`.

```bash
cooler dump --join {input.mcool}::resolutions/{binsize} | cooler load --format bg2 {new.chrom.sizes}:{binsize} - {output.cool}
cooler zoomify --balance {output.cool}
```

**Comment by SergejN** [comment]
thanks! this is very helpful and will save me a lot of time :)

**Comment by lauradmartens** [comment]
Thanks a lot! I agree, it saved a lot of time ;)

**Comment by golobor** [comment]
i wonder if tricks like this one deserve an ipynb and/or a page in sphinx
docs?..
